### PR TITLE
feat(job): add trace column to gitlab_project_job table

### PR DESCRIPTION
**Pull request by pdecat**
This PR adds a `trace` column to the `gitlab_project_job` table.

This can be useful to search for keywords in job logs, e.g.:

```
> select
  id, name, started_at
from
  gitlab_project_job
where
  project_id = '6666' and name = 'myjob' and trace like '%deadlock%'
limit
  10;
```

### CODE REVIEW

File: gitlab/table_project_job.go [hunk 1]
```diff
@@ -3,12 +3,20 @@ package gitlab
...
 	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
 	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
 	api "github.com/xanzy/go-gitlab"
 )
 
+type ProjectJob struct {
+	api.Job
+	Trace *string
+}
+
```
Comment (graza-io) [inline gitlab/table_project_job.go:19]: Is this required, we seemingly only use it to pass `nil` for trace as it's obtained via a hydrate call anyway?

File: gitlab/table_project_job.go [hunk 2]
```diff
@@ -45,7 +53,7 @@ func listProjectJobs(ctx context.Context, d *plugin.QueryData, h *plugin.Hydrate
 		}
 
 		for _, job := range jobs {
-			d.StreamListItem(ctx, job)
+			d.StreamListItem(ctx, ProjectJob{*job, nil})
```
Comment (graza-io) [inline gitlab/table_project_job.go:56]: Does this bring any value?

File: gitlab/table_project_job.go [hunk 3]
```diff
@@ -231,12 +273,19 @@ func projectJobColumns() []*plugin.Column {
...
 			Name:        "tag",
 			Type:        proto.ColumnType_BOOL,
 			Description: "Indicates if the job was started by a tag.",
 		},
+		{
+			Name:        "trace",
+			Type:        proto.ColumnType_STRING,
+			Description: "The trace (aka log) of the job.",
+			Hydrate:     getProjectJobTrace,
+			Transform:   transform.FromValue(),
```
Comment (graza-io) [inline gitlab/table_project_job.go:288]: 👍🏻

**Review by graza-io — CHANGES_REQUESTED** [review]
Please see comments.

File: gitlab/table_project_job.go (hunk 1 above)
Comment (pdecat) [reply gitlab/table_project_job.go:19]: Indeed, that comes from my initial tests where I planned to enrich this struct with the data retrieved from the trace API.

Cleanup in progress...

Comment (pdecat) [reply gitlab/table_project_job.go]: Cleaned that up.

File: gitlab/table_project_job.go (hunk 2 above)
Comment (pdecat) [reply gitlab/table_project_job.go]: Cleaned that up.

**Pull request merged**